src/Fixer/LanguageConstruct/NullableTypeDeclarationFixer.php

```diff
@@ -0,0 +1,326 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\LanguageConstruct;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Analyzer\Analysis\TypeAnalysis;
+use PhpCsFixer\Tokenizer\Analyzer\FunctionsAnalyzer;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @author John Paul E. Balandan, CPA <paulbalandan@gmail.com>
+ */
+final class NullableTypeDeclarationFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    private const OPTION_USE_EXPLICIT_NULL = 'use_explicit_null';
+    private const OPTION_USE_QUESTION_MARK = 'use_question_mark';
+
+    private int $candidateTokenKind;
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Single base type declaration should be marked nullable by using the configured syntax.',
```

Comment:
```suggestion
            'Nullable type declaration should be standardised using configured syntax.',
```

Reasoning:
- I don't understand what "Single base type declaration" is 😅 
- "should be marked nullable" is wrong, because it's already marked, fixer just converts to desired syntax (if needed)

Reply:
I actually lifted this from the PHP docs:
> ### Nullable type syntactic sugar
> A single base type declaration can be marked nullable by prefixing the type with a question mark (?). Thus `?T` and `T|null` are identical.

https://www.php.net/manual/en/language.types.declarations.php

Reply:
But it refers to different scenario. "Single base type declaration" is e.g. `Foo`, and it "can be marked as nullable" either with `?` or `|null`. From this fixer's perspective type is already marked as nullable (hence it's not "single" type anymore).

Reply:
Okay. But I think we should narrow down `nullable type declaration` as it can also refer to three or more atomic types unioned, which this fixer does not fix.

Reply:
@paulbalandan I've edited suggestion, let's use "nullable single type declaration". Can you rebase and resolve conflicts, so we can finish this useful fixer?